src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -7143,6 +7172,52 @@ def _update_artifact_type(params):
     logger.warning("Deployment type: %s. To override deployment type, please specify the --type parameter. "
                    "Possible values: war, jar, ear, zip, startup, script, static", params.artifact_type)
 
+def _get_instances_list_arm_internal(params):
+    url = _build_get_instances_list_arm_url(params)
+    response = send_raw_request(params.cmd.cli_ctx, "GET", url)
+    if response.status_code == 200:
+        logger.info("Instances list retrieved successfully.")
+        return response.json().get("value", [])
+    raise CLIError("Unable to get instances list.")
+
+def _get_instance_id_internal(param):
+    instances = _get_instances_list_arm_internal(param)
+
+    if not instances or len(instances) == 0:
+        raise CLIError("No instances found.")
+    
+    logger.info("returning instance id for instance: %s", instances[0]['name'])
+    return instances[0]['name']
+
+
+def _get_affinity_cookie_internal(param):
+    instance = _get_instance_id_internal(param)
+    logger.info("returning affinity cookie for instance: %s", instance)
+    cookie = {
+        "ARRAffinity": instance,
+        "ARRAffinitySameSite": instance
+        }
+    return cookie
+
+def _warmup_kudu_and_get_cookie_internal(param):
+    scm_url = _get_scm_url(param.cmd, param.resource_group_name, param.webapp_name, param.slot)
+    cookie = _get_affinity_cookie_internal(param)
+    Max_Retries = 3
+    TimeOut = 300
+    SleepInterval = 60
+    for retry in range(Max_Retries):
```

Comment:
nit: you can also explore using `@retryable_method`
https://github.com/Azure/azure-cli/blob/95f16fb5d47a57aeaf8dadf9ad0a47625694347f/src/azure-cli/azure/cli/command_modules/appservice/custom.py#L1538